csharp/Directory.Build.props

```diff
@@ -29,7 +29,7 @@
     <Product>Apache Arrow ADBC library</Product>
     <Copyright>Copyright 2022-2024 The Apache Software Foundation</Copyright>
```

Comment:
Should we update the copyright years as a separate change?

Reply:
Oops. Yes, let me follow up with another PR.